cosmos/ingestion/ingest/ingest.py

```diff
@@ -196,164 +199,193 @@ def raise_timeout(var1, var2):
-            ' '.join(content.split()[:label_length])
-            if (pp_cls in classes) and (pp_score >= threshold)
-            else None
-            for content, pp_cls, pp_score in zip(doc_df['content'],
-                                                 doc_df['postprocess_cls'],
-                                                 doc_df['postprocess_score'])
-        ]
-
-        # if table_label series has no values, just return original df
-        if len(doc_df[doc_df["table_label"].astype(bool)]) == 0:
-            return original_df
 
-        # aggregate all doc words into one list of words
-        words = ' '.join(doc_df.content.tolist()).split()
+        # aggregate all doc words into one list of words, excepting tables and table captions
+        doc_df = doc_and_tables_dfs[0]
+        pp_classes_to_ignore = ['Table', 'Table Caption']
+        all_doc_words = []
+        for index, row in doc_df.iterrows():
+            if row['postprocess_cls'] not in pp_classes_to_ignore:
+                all_doc_words.append(row['content'].split())
 
-        # mark where each label occurs in list of all doc words
-        doc_df['indices'] = [
-            [
-                j
-                for j in range(len(words))
-                if table_label == ' '.join(words[j:j + label_length])
-            ]
-            for table_label in doc_df['table_label']
-        ]
+        context_column = 'content_enriched'
 
-        # iterate over list in indices column to extract words before labels in content
-        logger.info('getting semantic context')
-        doc_df['prefix'] = [
-            [
-                ' '.join(words[i - spans:i])
-                if i - spans >= 0
-                else ' '.join(words[0:i])
-                for i in index
-            ]
-            for index in doc_df['indices']
-        ]
+        tables_df = doc_and_tables_dfs[1]
+        tables_df[context_column] = None  # create enriched content column
+        original_tables_df = tables_df.copy()  # copy to iterate over, copy to update
+        tables_df_columns = tables_df.columns.tolist()  # columns for writing final df
+        logger.debug(f'tables_df_columns:{tables_df_columns}')
+        pdf_name = original_tables_df.pdf_name.to_list()[0]
 
-        # iterate over list in indices column to extract words after labels in content
-        doc_df['suffix'] = [
-            [
-                ' '.join(words[i + label_length:i + label_length + spans])
-                if i + label_length + spans <= len(words)
-                else ' '.join(words[i + label_length:len(words)])
-                for i in index
+        label_length = 2
+        # attempt to update empty 'caption_contents' from 'content'
+        try:
+            original_tables_df['caption_content'] = [
+                re.sub(r"[^0-9a-zA-Z ]+", '', ' '.join(content.split()[:label_length]))
+                if (caption_content is None) and
+                   (pp_score >= threshold) and
```

Comment:
Can you add a similar filter for `base_confidence` with a separate (but defaulting to the same) value?

Reply:
Not sure what base_confidence is here, what's the goal?

Reply:
Basically it's another handle on filtering to only the best-quality table candidates. `base_confidence` is the certainty score that comes out of the visual detection algorithm. `postprocess_score` is the score that comes out of the separate xgboost postprocessing module.

Reply:
OK sounds fine to me. I can only see the columns below in the _tables.parquet though. Should base_confidence be present there, or elsewhere?

1. 'pdf_name',
2.  'dataset_id',
3.  'caption_content',
4.  'caption_page',
5.  'caption_bb',
6.  'pdf_dims',
7.  'detect_score',
8.  'postprocess_score',
9.  'content',
10.  'obj_page',
11.  'obj_bbs',
12.  'img_pth',
13.  'content_enriched'